test/success/J.prl

```diff
@@ -0,0 +1,50 @@
+Thm Connection(#l:lvl) : [
+  (->
+   [ty : (U #l kan)]
+   [a b : ty]
+   [p : (path [_] ty a b)]
+   (path [i] (path [_] ty a (@ p i)) (abs [_] a) p))
+] by [
+  lam ty, a, b, p.
+    <i j>
+      `(hcom 0~>1 ty a
+        [i=0 [k]
+         (hcom 1~>0 ty (@ p k)
+          [k=0 [_] a]
+          [k=1 [l] (@ p l)])]
+        [i=1 [k]
+         (hcom 1~>j ty (@ p k)
+          [k=0 [_] a]
+          [k=1 [l] (@ p l)])]
+        [j=0 [k]
+         (hcom 1~>0 ty (@ p k)
+          [k=0 [_] a]
+          [k=1 [l] (@ p l)])]
+        [j=1 [k]
+         (hcom 1~>i ty (@ p k)
+          [k=0 [_] a]
+          [k=1 [l] (@ p l)])]
+        [i=j [k]
+         (hcom 1~>i ty (@ p k)
+          [k=0 [_] a]
+          [k=1 [l] (@ p l)])])
+].
+
+Thm J(#l:lvl) : [
+  (->
+   [ty : (U #l kan)]
+   [a : ty]
+   [fam : (-> [x : ty] (path [_] ty a x) (U #l kan))]
+   [d : ($ fam a (abs [_] a))]
+   [x : ty]
+   [p : (path [_] ty a x)]
+   ($ fam x p))
+] by [
+  lam ty, a, fam, d, x, p.
+    `(coe 0~>1
+      [i] ($ fam (@ p i) (@ ($ (Connection #l) ty a x p) i))
```

Comment:
I believe you do not need connection, and can prove this by
```
($ fam (whatever to make this type check) (@ ($ (a hcom from refl with one side being refl and the other side being p) ty a x p) i))
```

Reply:
Cool! I'll try your idea

Reply:
Hmm, I couldn't figure out how to make it work, at least not immediately.

Reply:
Something like this? Don't really want to switch branches now.
```
Thm J(#l:lvl) : [
  (->
   [ty : (U #l kan)]
   [a : ty]
   [fam : (-> [x : ty] (path [_] ty a x) (U #l kan))]
   [d : ($ fam a (abs [_] a))]
   [x : ty]
   [p : (path [_] ty a x)]
   ($ fam x p))
] by [
  lam ty, a, fam, d, x, p.
    `(coe 0~>1
      [i] ($ fam (hcom 0~>1 ty a [i=0 [_] a] [i=1 [j] (@ p j)])
                 (abs [j] (hcom 0~>j ty a [i=0 [_] a] [i=1 [j] (@ p j)])))
      d)
].
```

Reply:
@favonia Wow, nicely done! My composition-fu is not yet at that level 😄